output-comparator/src/main/java/org/mobilitydata/gtfsvalidator/outputcomparator/io/NoticeStat.java

```diff
@@ -0,0 +1,133 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Used to store details about a particular {@code ValidationNotice}. For one {@code
+ * ValidationNotice} this gives information about:
+ *
+ * <ul>
+ *   <li>the number of sources that raised this {@code ValidationNotice}
+ *   <li>the ids of the sources that raised this {@code ValidationNotice} and the url to be used to
+ *       download them
+ *   <li>the total number of this {@code ValidationNotice} in each source concerned
+ * </ul>
+ */
+public class NoticeStat {
+
+  protected static final String AFFECTED_SOURCES_COUNT = "affectedSourcesCount";
+  protected static final String AFFECTED_SOURCES = "affectedSources";
+  protected static final String COUNT_PER_SOURCE = "countPerSource";
+  public static final String URL_PATTERN =
+      "https://storage.googleapis.com/storage/v1/b/%s_archives_\\d{4}-\\d{2}-\\d{2}/o/\\w+.zip\\?alt=media";
+  private final SortedMap<String, String> affectedSources;
+  private final SortedMap<String, Integer> countPerSource;
+  private int affectedSourcesCount;
+
+  public NoticeStat(
+      int affectedSourcesCount,
+      SortedMap<String, String> affectedSourced,
+      SortedMap<String, Integer> countPerSource) {
+    this.affectedSourcesCount = affectedSourcesCount;
+    this.affectedSources = affectedSourced;
+    this.countPerSource = countPerSource;
+  }
+
+  public NoticeStat() {
+    this(0, new TreeMap<>(), new TreeMap<>());
+  }
+
+  @VisibleForTesting
+  public SortedMap<String, String> getAffectedSources() {
+    return affectedSources;
+  }
+
+  @VisibleForTesting
+  public Map<String, Integer> getCountPerSource() {
+    return countPerSource;
+  }
+
+  @VisibleForTesting
+  public int getAffectedSourcesCount() {
+    return affectedSourcesCount;
+  }
+
+  public static String retrieveSourceUrl(String urlAsString, String sourceId) {
+    Pattern pattern = Pattern.compile(String.format(URL_PATTERN, sourceId));
+    Matcher matcher = pattern.matcher(urlAsString);
+    matcher.find();
+    return matcher.group();
+  }
+
+  /**
+   * Updates field countPerSource for a given sourceId
+   *
+   * @param sourceId the id of the source to update
+   * @param newCount the new value for {@code NoticeStat#count}
+   */
+  private void updateCountPerSource(String sourceId, Integer newCount) {
+    Integer currentCount = this.countPerSource.getOrDefault(sourceId, 0);
+    this.countPerSource.put(sourceId, currentCount + newCount);
+  }
+
+  /**
+   * Updates all fields of this {@code NoticeStat}.
+   *
+   * @param sourceId the id of the source
+   * @param noticeCount the number of notices raised by the latest dataset version from a given
+   *     source identified by its id
+   */
+  public void update(String sourceId, int noticeCount, String urls) {
+    this.affectedSources.put(sourceId, retrieveSourceUrl(urls, sourceId));
+    updateCountPerSource(sourceId, noticeCount);
+    this.affectedSourcesCount = this.affectedSources.size();
```

Comment:
Do you need to store the value or you can use `this.affectedSources.size()` during serialization?

Reply:
Indeed, this is more simple!